channel/pools.go

```diff
@@ -813,8 +815,40 @@ func (pool *dropPool) eraseDrops() {
 	pool.drops = make(map[int32]fieldDrop)
 }
 
-func (pool *dropPool) playerAttemptPickup(dropID int32, position pos) (bool, item) {
-	return false, item{}
+func (pool *dropPool) getPickupDrop(dropID int32) (error, fieldDrop) {
+	drop := pool.findDropFromID(dropID)
+
+	if reflect.DeepEqual(drop, fieldDrop{}) {
```

Comment:
reflect.DeepEqual is expensive. Update findDropFromID to return the drop and and ok value.